PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -97,68 +112,138 @@ struct TaskPolarisationCharmHadrons {
   template <uint8_t channel, bool withMl, typename Cand>
   void runPolarisationAnalysis(Cand const& candidate, Tracks const& tracks)
   {
-    float pxDau{-1000.}, pyDau{-1000.}, pzDau{-1000.};
-    float pxCharmHad{-1000.}, pyCharmHad{-1000.}, pzCharmHad{-1000.};
-    float massDau{0.}, invMassCharmHad{0.}, invMassCharmHadForSparse{0.};
-    float rapidity{-999.};
-    std::array<float, 3> outputMl{};
-    if constexpr (channel == charm_polarisation::DecayChannel::DstarToDzeroPi) {
-      pxDau = candidate.pxSoftPi();
-      pyDau = candidate.pySoftPi();
-      pzDau = candidate.pzSoftPi();
-      pxCharmHad = candidate.pxDstar();
-      pyCharmHad = candidate.pyDstar();
-      pzCharmHad = candidate.pzDstar();
-      massDau = massPi;
-      auto prongSoftPi = candidate.template prongPi_as<aod::Tracks>();
-      invMassCharmHad = (prongSoftPi.sign() > 0) ? candidate.invMassDstar() : candidate.invMassAntiDstar();
-      invMassCharmHadForSparse = (prongSoftPi.sign() > 0) ? (invMassCharmHad - candidate.invMassD0()) : (invMassCharmHad - candidate.invMassD0Bar()); // different for D*
-      rapidity = candidate.y(massDstar);
+
+    // inv. mass hypothesis to loop over
+    // e.g.: Lc->pKpi has the ambiguity pKpi vs. piKp
+    uint8_t nMassHypos = (channel == charm_polarisation::DecayChannel::LcToPKPi) ? charm_polarisation::MassHyposLcToPKPi::NMassHypoLcToPKPi : 1;
```

Comment:
No need to check for every candidate, it can be set in the `init` function

Reply:
but this would mean to have `channel` as configurable somewhere, right?

Reply:
addressed